extensions/wikia/AdEngine/js/lookup/lookupFactory.js

```diff
@@ -122,6 +122,25 @@ define('ext.wikia.adEngine.lookup.lookupFactory', [
 			});
 		}
 
+		function waitForResponse(milisToTimeout) {
+			return waitForOrTimeout(function (resolve) {
+				if (hasResponse()) {
+					resolve();
+				} else {
+					addResponseListener(resolve);
+				}
+			}, milisToTimeout);
+		}
+
+		function waitForOrTimeout(f, milisToTimeout) {
+			milisToTimeout = milisToTimeout || 2000;
+			var timeout = new Promise(function (resolve, reject) {
+					setTimeout(reject, milisToTimeout);
+				});
+
+			return Promise.race([new Promise(f), timeout]);
```

Comment:
Promise is not supported by IE11 :trollface:

Reply:
http://community.wikia.com/wiki/Help:Supported_browsers :trollface: